### ISSUE TITLE
Enums to pointer to enums

### ISSUE THREAD

**Issue by ThePedroo** [Cogmasters/concord#176]
## Describe the feature
This feature request is to the change of pure enums to pointer to enums.

## Acceptance criteria
- [ ] All enums of structures of gencodecs must use pointer to enums instead.

## Additional context

Concord has been using enums for some time, and they have some issues when it comes to identifying whether or not the field was provided, as they would always default to 0. Especially in cases to send requests, where the enum field is nullable, it would be problematic.

**Comment by lcsmuller** [comment]
> they have some issues when it comes to identifying whether or not the field was provided, as they would always default to 0

@ThePedroo Sorry for reviving a 2-year-old ticket, but wouldn't this be an issue with any primitives, not just enums? Meaning, any primitive would need to be converted into a pointer-to-primitive

**Comment by ThePedroo** [comment]
Not really. Others, unlike flags, have no difference between 0 and undefined. At least none I'm aware of.
